File: loader/src/test/scala/com/hurence/historian/model/MeasureTest.scala
package com.hurence.historian.model

import java.util.Date

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.Calendar

import com.hurence.timeseries.model.Measure

class MeasureTest {

 /* @Test
  def testMeasureV0() = {

    val cal = Calendar.getInstance

    val measureName = "cpu"
    val measureValue = 76.4d
    val measureTimestamp = new Date().getTime
    val measureYear = cal.get(Calendar.YEAR)
    val measureMonth = cal.get(Calendar.MONTH)
    val measureLon = "5.373523"
    val tags = Map("host" -> "host1", "ip" -> "12.1.1.1", "lon" -> measureLon)

    import scala.collection.JavaConversions._
    import scala.collection.JavaConverters._

    val measure = new Measure(measureName, measureValue, measureTimestamp, measureYear, measureMonth, "2020-15-04", tags)

    assertEquals(measureName, measure.getName)
    assertEquals(measureValue, measure.getValue)
    assertEquals(measureTimestamp, measure.getTimestamp)
    assertEquals(measureYear, measure.getYear)
    assertEquals(tags.asJava, measure.getTags)
    assertEquals(measureLon, measure.getTags.get("lon"))

  }*/
}

File: core/src/test/java/com/hurence/timeseries/model/MeasureTest.java
package com.hurence.timeseries.model;

import org.junit.Test;

import java.util.*;

import static org.junit.Assert.*;


public class MeasureTest {

  @Test
  public void testMeasure() {

    String measureName = "cpu";
    double measureValue = 76.4d;
    long measureTimestamp = 223849510000L;
    String measureLon = "5.373523";
    float measureQuality = 98.0f;
    String measureDay = "1977-02-03";
    Map<String,String> tags = new HashMap<>();
    tags.put("host" , "host1");
    tags.put("ip" ,"12.1.1.1");
    tags.put("lon" ,measureLon);

    Measure measure = Measure.builder()
      .name(measureName)
      .value(measureValue)
      .timestamp(measureTimestamp)
      .quality(measureQuality)
      .tags(tags)
      .build();

    assertEquals(measureName, measure.getName());
    assertEquals(measureValue, measure.getValue(), 0.0f);
    assertEquals(measureTimestamp, measure.getTimestamp());
    assertEquals(measureDay, measure.getDay());
    assertEquals(tags, measure.getTags());
    assertEquals(measureLon, measure.getTags().get("lon"));
    assertEquals(measureLon, measure.getTag("lon"));
    assertNull(measure.getTag("unknown tag"));
    assertTrue(measure.containsTag("host"));
  }
}
